File: modules/AcroStack.OpenIddictManagement/Dtos/CreateOpenIddictApplicationDto.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace AcroStack.OpenIddictManagement;

/// <summary>
/// DTO for creating a new OpenIddict application.
/// Mirrors <c>OpenIddictApplicationDescriptor</c> fields. The
/// <c>ClientSecret</c> is hashed by <c>IAbpApplicationManager</c> on save.
/// </summary>
public class CreateOpenIddictApplicationDto
{
    [Required]
    [StringLength(64)]
    public string ClientId { get; set; } = string.Empty;

    [StringLength(200)]
    public string? DisplayName { get; set; }

    /// <summary>One of <c>OpenIddictConstants.ClientTypes</c> (public/confidential).</summary>
    public string? ClientType { get; set; }

    /// <summary>One of <c>OpenIddictConstants.ConsentTypes</c>.</summary>
    public string? ConsentType { get; set; }

    /// <summary>Plain-text secret; hashed on save. Null for public clients.</summary>
    public string? ClientSecret { get; set; }

    public List<string> Permissions { get; set; } = new();

    [MaxLength(20)]
    public List<string> RedirectUris { get; set; } = new();

    [MaxLength(20)]
    public List<string> PostLogoutRedirectUris { get; set; } = new();

    public List<string> Requirements { get; set; } = new();
}

File: react/src/api/models/acroStack/openIddictManagement/CreateOpenIddictApplicationDto.ts
/* oxlint-disable */

/**
 * @description DTO for creating a new OpenIddict application.\r\nMirrors `OpenIddictApplicationDescriptor` fields. The\r\n`ClientSecret` is hashed by `IAbpApplicationManager` on save.
 * @type object
 */
export type AcroStackOpenIddictManagementCreateOpenIddictApplicationDto = {
  /**
   * @minLength 0
   * @maxLength 64
   * @type string
   */
  clientId: string;
  /**
   * @minLength 0
   * @maxLength 200
   * @type string | undefined
   */
  displayName?: string | null;
  /**
   * @description One of `OpenIddictConstants.ClientTypes` (public/confidential).
   * @type string | undefined
   */
  clientType?: string | null;
  /**
   * @description One of `OpenIddictConstants.ConsentTypes`.
   * @type string | undefined
   */
  consentType?: string | null;
  /**
   * @description Plain-text secret; hashed on save. Null for public clients.
   * @type string | undefined
   */
  clientSecret?: string | null;
  permissions?: string[] | null;
  redirectUris?: string[] | null;
  postLogoutRedirectUris?: string[] | null;
  requirements?: string[] | null;
};
